### PR TITLE
Add function traces and logs

### PR DESCRIPTION
**Pull request by johscheuer**

### CODE REVIEW

File: step_5/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Function Spans, Logs and Tags
+
+In this part we show the advantage of using tags and how to trace function.
```
Comment (maxbischoff) [inline step_5/README.md]:
```suggestion
In this part we show the advantage of using tags and how to trace functions.
```

Or `how to trace a function`

File: step_5/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Also the Toast backend service will return sometimes an emtpy slice.
```
Comment (maxbischoff) [inline step_5/README.md]:
```suggestion
Also the Toast backend service will sometimes return an emtpy slice.
```

File: step_5/backend/main.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	defer span.Finish()
+
+	// Set tags for filtering
+	ext.SpanKindRPCServer.Set(span)
+	ext.HTTPUrl.Set(span, r.URL.String())
+	ext.HTTPMethod.Set(span, r.Method)
+
+	w.Header().Set("Content-Type", "application/json")
+
+	if rand.Int()%4 != 0 {
```
Comment (maxbischoff) [inline step_5/backend/main.go:39]:
```suggestion
        // return the toast slice 3 out of 4 calls, else return a HTTP 500
	if rand.Int()%4 != 0 {
```

Please add a comment on the intent of this rand calculation

**Review by maxbischoff — CHANGES_REQUESTED** [review]
Some minor things. I would also consider splitting this step into two to show function traces and then logs (or vice versa), but it is also fine as-is IMO

**Review by johscheuer — COMMENTED** [review]
IMHO I don't see a huge benefit in splitting this up even further since both changes are pretty simple.

**Pull request merged**